src/main/Main.hs

```diff
@@ -291,6 +292,10 @@ commandLineHandler currentDir progName isInterpreter = complicatedOptions
                     "Print out handy path information"
                     pathCmd
                     Stack.Path.pathParser
+        addCommand' "ls"
```

Comment:
I think this should be `list-snapshots`.  There is already some precedent here, with `list-dependencies`.

How about the default for `stack list-snapshots` first displaying the remote snapshots, and then displaying the local snapshots.  Then, can have `stack list-snapshots remote` and `stack list-snapshots local`, particularly useful for when a script wants to access this info.

Reply:
@mgsloan As mentioned by Harendra [here](https://github.com/commercialhaskell/stack/pull/3252#issuecomment-345688530), `ls` is the umberlla command. `dependencies` will be made to work under this interface ( I can work on this - after the PR get's merged)

> How about the default for stack list-snapshots first displaying the remote snapshots, and then displaying the local snapshots. Then, can have stack list-snapshots remote and stack list-snapshots local, particularly useful for when a script wants to access this info.

Display `remote` snapshots involves an HTTP call and I want to avoid doing that by default. By default, I'm planning to show the `local` snapshots and then have different options for displaying each of them. Let me know if you think otherwise.

Reply:
Ah, I had not read those comments recently.  Ok, I am fine with doing it that way.